packages/headless/src/features/case-fields/case-fields-slice.ts

```diff
@@ -0,0 +1,51 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {
+  disableCaseClassifications,
+  enableCaseClassifications,
+  fetchCaseClassifications,
+  setCaseField,
+} from './case-fields-actions';
+import {getCaseFieldsInitialState} from './case-fields-state';
+
+export const caseFieldsReducer = createReducer(
+  getCaseFieldsInitialState(),
+
+  (builder) => {
+    builder
+      .addCase(setCaseField, (state, action) => {
+        const {fieldName, fieldValue} = action.payload;
+        const field = state.fields[fieldName];
+        if (field) {
+          field.value = fieldValue;
+        } else {
+          state.fields[fieldName] = {
+            value: fieldValue,
+            suggestions: [],
+          };
+        }
+      })
+      .addCase(enableCaseClassifications, (state) => {
```

Comment:
What are the scenarios where you would want to enable or disable case classification, and what impact should it have on the internal logic (if any) ?

For example, if case assis is disabled, and someone dispatch `fetchCaseClassifications` action, what should happen ?

Right now I'm not sure I understand what the `enabled` flag does exactly.

Reply:
So this "enabled" flag activates the **automatic** retrieval of case classifications when changes are made to the the `caseInput` values. You can still disable it and fetch by dispatching the fetch action but the intended way of using it is to register caseInputs (ex: subject and description) and instantiate a case classification controller that enables this automatic retrieval feature and will react to changes in the inputs.

Reply:
It might be simpler to manage this UX detail outside of headless, at the UI level.

Generally speaking, we have not embedded UX behaviour at the reducer level. When we have, it's been opinionated with no ability to configure. e.g. clicking a query suggestion updates the search box q.

As long as headless provide the building blocks to a) adjust an input value, and b) trigger a fetch, an implementer can combine them as they wish. This removes the need for
1. Implementers to specify config into headless.
2. Coveo devs to maintain the logic, which can spiral in complexity over time as more options are added to case-assist. Less options delays questions like "what should the behaviour be if option A is `X` and option B is `Y`".


```
<input onChange: autoUpdateClassifications/>

function autoUpdateClassifications(e) {
   inputController.updateValue(e.value);
   inputController.fetchClassifications();
}
```

Reply:
As @nathanlb mentioned, the idea behind the "enabled" flags is to make it easier to the implementer by retrieving the document and field suggestions automatically. We can decide to remove the flag and let the implementer invoke the actions himself, but it adds more complexity on the implementer side. I'm worried about implementers forgetting to update suggestions.

Reply:
Going to play devil's advocate, the phrase "worried about implementers forgetting" sounds like there is an opinionated UX we want to push devs towards?

If so, we can do this at the controller level, where e.g. the `updateValue` method dispatches the fetch, similar to the facet controllers. This could make sense if it's the behaviour we expect 99% of implementations to use.

For the 1% that does not want this, the door remains open to override the method and dispatch only the action that updates the input value.

If it's closer to 50-50, then being opinionated would not be the best path.
However, providing the building blocks for devs to compose still might have benefits of minimizing config and code to maintain.

Reply:
The planned behaviour is to have the caseInput controller fetch when updateValue is called, but only if the enabled flag is on. The purpose of there being a flag is also that the case form and the case field with case classifications might be on distinct steps of the case submission form. That's how it'll be configured in our cookbook implementation. In that case we don't want classifications to be fetched if there's no use for them yet.

Reply:
@samisayegh we'll follow your advice and remove the enable/disable pattern for now. Like I mentioned we'll be putting together a case assist implementation as part of our cookbook and so if the DX reveals some pain points we can re-advise. It would've been perhaps a little more efficient to settle this while reviewing the design document so hopefully we surface these types of questions earlier next time. Thanks for the review! 😁